dandischema/datacite.py

```diff
@@ -0,0 +1,192 @@
+from copy import deepcopy
+import typing as ty
+
+from .models import Organization, Person, PublishedDandiset, RoleType
+
+DATACITE_CONTRTYPE = {
+    "ContactPerson",
+    "DataCollector",
+    "DataCurator",
+    "DataManager",
+    "Distributor",
+    "Editor",
+    "HostingInstitution",
+    "Producer",
+    "ProjectLeader",
+    "ProjectManager",
+    "ProjectMember",
+    "RegistrationAgency",
+    "RegistrationAuthority",
+    "RelatedPerson",
+    "Researcher",
+    "ResearchGroup",
+    "RightsHolder",
+    "Sponsor",
+    "Supervisor",
+    "WorkPackageLeader",
+    "Other",
+}
+
+
+DATACITE_IDENTYPE = {
+    "ARK",
+    "arXiv",
+    "bibcode",
+    "DOI",
+    "EAN13",
+    "EISSN",
+    "Handle",
+    "IGSN",
+    "ISBN",
+    "ISSN",
+    "ISTC",
+    "LISSN",
+    "LSID",
+    "PMID",
+    "PURL",
+    "UPC",
+    "URL",
+    "URN",
+    "w3id",
+}
+DATACITE_MAP = {el.lower(): el for el in DATACITE_IDENTYPE}
+
+
+def to_datacite(meta: ty.Union[dict, PublishedDandiset]) -> dict:
+    """Convert published Dandiset metadata to Datacite"""
+    if not isinstance(meta, PublishedDandiset):
+        meta = PublishedDandiset(**meta)
+
+    attributes = {}
+    attributes["identifiers"] = [
+        # TODO: the first element is ignored, not sure how to fix it...
+        {"identifier": f"https://doi.org/{meta.doi}", "identifierType": "DOI"},
+        {
+            "identifier": f"https://identifiers.org/{meta.id}",
+            "identifierType": "URL",
+        },
+        {
+            "identifier": f"https://dandiarchive.org/dandiset/{meta.identifier}",
+            "identifierType": "URL",
+        },
+    ]
+
+    attributes["doi"] = meta.doi
+    attributes["titles"] = [{"title": meta.name}]
+    attributes["descriptions"] = [
+        {"description": meta.description, "descriptionType": "Abstract"}
+    ]
+    attributes["publisher"] = "DANDI Archive"
+    attributes["publicationYear"] = str(meta.datePublished.year)
+    # not sure about it dandi-api had "resourceTypeGeneral": "NWB"
```

Comment:
note: lost track of comments... I was wondering there about where we could get specific  data standards -- we would have them in `AssetsSummary. dataStandard: Optional[List[StandardsType]] `, but not sure where we would stick them yet in datacite